File: hw2/program/parallelepiped.cpp
//------------------------------------------------------------------------------
// parallelepiped.cpp - содержит процедуры для работы с параллелепипедом
//------------------------------------------------------------------------------

#define _CRT_SECURE_NO_WARNINGS
#include "stdio.h"
#include "parallelepiped.h"

//------------------------------------------------------------------------------
// Ввод параметров параллелепипеда из потока
void Parallelepiped::In(FILE* file) {
    fscanf(file, "%d", &a);
    fscanf(file, "%d", &b);
    fscanf(file, "%d", &c);
}

// Случайный ввод параметров параллелепипеда
void Parallelepiped::InRnd() {
    a = Shape::rnd20.Get();
    b = Shape::rnd20.Get();
    c = Shape::rnd20.Get();
}

//------------------------------------------------------------------------------
// Вывод параметров параллелепипеда в поток
void Parallelepiped::Out(FILE* file) {
    fprintf(file, "It is Parallelepiped. First edge: a = %d. Second edge: b = %d. Third edge: c = %d. Surface area = %f. Material density: %lf\n",
        a, b, c, SurfaceArea(), materialDencity);
}

//------------------------------------------------------------------------------
// Вычисление площади поверхности параллелепипеда
double Parallelepiped::SurfaceArea() {
    return double(2 * (a * b + b * c + a * c));
}

File: hw3/program/parallelepiped.py
import random
from shape import *
# ----------------------------------------------


class Parallelepiped(Shape):
    def __init__(self):
        self.a = 0
        self.b = 0
        self.c = 0
        self.material_density = 0.0

    def get_figure_from_file(self, data, i):
        self.material_density = float(data[i])
        self.a = int(data[i + 1])
        self.b = int(data[i + 2])
        self.c = int(data[i + 3])

    def get_random_figure(self):
        self.a = random.randint(1, 101)
        self.b = random.randint(1, 101)
        self.c = random.randint(1, 101)
        self.material_density = random.uniform(1, 10001)

    def surface_area(self):
        return float(2 * (self.a * self.b + self.b * self.c + self.a * self.c))

    def write(self, output):
        output.write("It is Parallelepiped: First edge: a = {}. Second edge: b = {}. Third edge: c = {}. "
                     "Surface area = {}. Material density = {}"
                     .format(self.a, self.b, self.c, self.surface_area(), self.material_density))
